### PR TITLE
Fix for URL path handling on Windows. (Fixes #266)

### LINKED ISSUE #266: "Illegal character in path" on Windows
**What version of the credhub server you are using?**
main Branch

**What version of the credhub cli you are using?**
2.9.3

**If you were attempting to accomplish a task, what was it you were attempting to do?**
Running credhub on Windows.

**What did you expect to happen?**
Credhub should start.

**What was the actual behavior?**
`IllegalArgumentException` on startup because of `Illegal character in path`, because of `Paths.get(string, string)` usage for URLs.

Reason: `Paths.get(string, string)` uses `\` on Windows instead of `/`. But `\` is illegal for URLs.

Solution: I will provide a pull request.

**Please confirm where necessary:**
* [ ] I have included a log output
* [ ] My log includes an error message
* [ ] I have included steps for reproduction

If you are a PCF customer with an Operation Manager (PCF Ops Manager) please direct your questions to support (https://support.pivotal.io/)

### PR DESCRIPTION
**Pull request by thomas-worm-datev**
This pull request fixes URL path handling on Windows.

### CODE REVIEW

**Comment by thomas-worm-datev** [comment]
This pull request fixes the [issue #266](https://github.com/cloudfoundry/credhub/issues/266).

**Review by bruce-ricard — CHANGES_REQUESTED** [review]
Hi @thomas-worm-datev , thank you for your PR.

Could you please add a test to your change please, to explain and describe what your change is about?

Thanks.

File: components/auth/src/main/kotlin/org/cloudfoundry/credhub/config/OAuthProperties.kt [hunk 1]
```diff
@@ -35,10 +35,16 @@ class OAuthProperties {
     private fun getResolvedUri(extension: String): URI {
         val authServer = if (internalUrl != null) internalUrl else url
         val base = URI(authServer!!)
-        val path = Paths.get(base.path, extension).toString()
+        val path = concatPath(base.path, extension).toString()
         return base.resolve(path)
     }
 
+    private fun concatPath(p1: String, p2: String): String {
```
[the remark this answers is not in the document]
Comment (strehle) [reply components/auth/src/main/kotlin/org/cloudfoundry/credhub/config/OAuthProperties.kt:42]: hi @bruce-ricard  I checked the usage and it is for credhub internal resolution of pathes to UAA so with 
https://github.com/cloudfoundry/credhub/pull/325/files it simply does same / similar this we have in UAA already in order to resolve pathes to OIDC endpoints

**Comment by hsinn0** [comment]
https://github.com/cloudfoundry/credhub/issues/266 already fixed and closed with another PR

**Pull request closed without merging**